lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,302 @@
+    #    - push said branch to the origin
+    #    - create a pull request for that branch
+    #
+    # More info can be found here on how the whole process works:
+    #
+    #   https://github.com/ManageIQ/manageiq-cross_repo-tests
+    #
+    # == Command structure
+    #
+    # @miq-bot run-tests [<repos-to-test>] [including <extra-repos>]
+    #
+    # where:
+    #
+    #   - a `repo` is of the form [org/]repo[@ref|#pr]
+    #   - `repos-to-test` is a list of repos to have tested
+    #   - `extra-repos` is a list of repos (gems) to override in the bundle
+    #
+    # Lists of repos should be comma delimited.
+    #
+    # == Examples
+    #
+    # === In ManageIQ/manageiq PR #1234:
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq-ui-classic
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=ManageIQ/manageiq#12345
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=ManageIQ/manageiq-ui-classic
+    #
+    #
+    # === In ManageIQ/manageiq PR #1234
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq-api,manageiq-ui-classic \
+    #     including Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature,ManageIQ/manageiq#1234
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=ManageIQ/manageiq-api
+    #   +  - TEST_REPO=ManageIQ/manageiq-ui-classic
+    #
+    # TODO:  Handle the "self" case, where `manageiq` is also a TEST_REPO
+    #
+    # (maybe include a "self" helper as well?)
+    #
+    # === In ManageIQ/manageiq-ui-classic PR #1234:
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq#5678
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=manageiq#5678
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=manageiq-ui-classic#1234
+    #
+    class RunTest < Base
+      # The user calling the command
+      attr_reader :issuer
+
+      # The org of the issue this command originated from
+      attr_reader :org
+
+      # The (extra) repo(s) being targeted to be included in the test run
+      attr_reader :repos
+
+      # The repo(s) that will have the test suite run
+      attr_reader :test_repos
+
+      # The *-cross_repo-tests rugged instance
+      attr_reader :rugged_repo
+
+      # The arguments for the `run-test` command being called
+      attr_reader :value
```

Comment:
:issue and :value are technically on every command (and :org is also from the base? I'm not seeing it assigned here), so I think it's better to move those down to the base class.  Though I do understand that could introduce potential issues in the other commands, so I'd be fine with a separate PR for that (perhaps separated and then merged before this one, so this one can use it?)

Reply:
> :issue and :value are technically on every command

@Fryguy `:issue` is, but `:value` is passed in as a param to `_execute`, and is not an `attr_reader`.  I added that just so I didn't have to pass it around, though, I think `:value` as a `attr_reader` might be able to go away now that I think about it... 🤔 

> (and :org is also from the base? I'm not seeing it assigned here)

I think I was considering using `:org` at one point, and then I guess I just did `issue.organization` in some places instead and called it good.  I can just remove that `attr_reader` now probably.